ods_ci/tests/Resources/Page/OCPDashboard/Monitoring/Metrics.robot

```diff
@@ -54,3 +55,42 @@ Get Query Results
         END
     END
     RETURN  ${metrics_query_result_row1_value}
+
+Run OpenShift Metrics Query
+    [Documentation]    Runs a query in the Monitoring section of Open Shift
+    ...    Note: in order to run this keyword OCP_ADMIN_USER.USERNAME needs to
+    ...    belong to a group with "view" role in OpenShift
+    ...    Example command to assign the role: oc adm policy add-cluster-role-to-group view rhods-admins
+    [Arguments]    ${query}    ${retry_attempts}=10    ${return_zero_if_result_empty}=False   ${usertype}=admin
+    Open OCP Console
+    IF    "${usertype}" == "basic"
+        LoginPage.Login To Openshift    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    ELSE
+        LoginPage.Login To Openshift    ${OCP_ADMIN_USER.USERNAME}    ${OCP_ADMIN_USER.PASSWORD}    ${OCP_ADMIN_USER.AUTH_TYPE}
+    END
+    OCPMenu.Switch To Administrator Perspective
+
+    # In OCP 4.9 metrics are under the Observe menu (it was called Monitoring in 4.8)
+    ${menu_observe_exists} =    Run Keyword and Return Status    Menu.Page Should Contain Menu    Observe
+    IF    ${menu_observe_exists}
+        Menu.Navigate To Page    Observe    Metrics
+    ELSE
+        ${menu_monitoring_exists} =    Run Keyword and Return Status    Menu.Page Should Contain Menu    Monitoring
+        IF    ${menu_monitoring_exists}
+            Menu.Navigate To Page    Monitoring    Metrics
+        ELSE
+            IF    "${usertype}" == "basic"
+                Fail
+                ...    msg=${TEST_USER.USERNAME} can't see the Observe/Monitoring section in OpenShift Console, please make sure it belongs to a group with "view" role
+            ELSE
+                Fail
+                ...    msg=${OCP_ADMIN_USER.USERNAME} can't see the Observe/Monitoring section in OpenShift Console, please make sure it belongs to a group with "view" role
+            END
+        END
+    END
+
+    Metrics.Verify Page Loaded
+    Metrics.Run Query    ${query}    ${retry_attempts}
+    ${result} =    Metrics.Get Query Results    return_zero_if_result_empty=${return_zero_if_result_empty}
+    RETURN    ${result}
```

Comment:
I'm not a big fan of using OCP UI navigation in our tests, it often gets broken over the various version we support. I think it should be safe enough for us to use the REST queries (we already have keywords for that)